ios/Classes/SwiftWorkmanagerPlugin.swift

```diff
@@ -28,27 +28,26 @@ public class SwiftWorkmanagerPlugin: FlutterPluginAppLifeCycleDelegate {
 extension SwiftWorkmanagerPlugin: FlutterPlugin {
     
     public static func register(with registrar: FlutterPluginRegistrar) {
+        
         let foregroundMethodChannel = FlutterMethodChannel(name: ForegroundMethodChannel.channelName, binaryMessenger: registrar.messenger())
         let instance = SwiftWorkmanagerPlugin()
         registrar.addMethodCallDelegate(instance, channel: foregroundMethodChannel)
         registrar.addApplicationDelegate(instance)
+        
     }
     
     public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
         
         switch call.method {
         case ForegroundMethodChannel.methods.initialize.rawValue:
             guard let callbackHandle: Int64 = call.arguments as? Int64 else {
-                result(WMPError.unexpectedMethodArguments(argumentsDescription: call.arguments.debugDescription))
+                result(WMPError.unexpectedMethodArguments(call.arguments.debugDescription))
```

Comment:
Is this necessary as in, when would this crash?
This could only happen if the plugin had bad code in it, in which case the whole plugin would be rendered useless.
I would rather just forcefully cast it to an Int.

(If this is what you are doing, I'm sorry I don't read Swift so well)

Reply:
It's more a question of best practices truly, I wouldn't force cast a value only for the sake of having to support an error scenario

Reply:
But it's like saying 

```
final s = "someString"

if (s is String) {
 final t = s as String;
}
```

It does not make sense. The Dart plugin will **always** send it as in `Int`

Reply:
I get your point, but if we go down this path, we could as well expect to never receive anything else than the `initialize` call methods in this channel, so we shouldn't verify the contents of `call.mehtod` either ? This doesn't look like clean code to me

Reply:
Sure why not. It make total sense for the iOS code imho

Reply:
Sorry, I cannot agree with you on this point ; not only is it dirty, but it also makes the code harder to understand. If we don't switch on the call's method to ensure we're indeed handling what we expect to handle, and we breakpoint in this method, it makes it much harder to understand where that comes from (especially since we won't have a stack trace, since this method is called by Flutter).
It also makes it harder to support adding new methods in the same channel.

Reply:
I'm not asking for you to remove the switch case. 
I just think the casting of arguments should be done safely.

Reply:
... Which it is, by "guard letting" the argument and its type ; force casting it seems like a lazy "I don't want to handle this error" move to me :-)

Reply:
No, why would your approach be safer? 
You know because it's your own code that this can only be an `Int`.
Now you swallow the error and throw it back to the user.

If you forcefully unwrap it here and Dart did not send an Int it will crash the app, which is actually more desirable.

Reply:
I just ran a couple scenarios ; one was force-casting the Int value sent by Dart through the MethodChannel to a String (causing a crash), the second was guard-letting it and returning a Flutter Error (as in the current implementation).
As a result, force-casting outputted "Lost connection to the device" without any clue on what caused the process to abort ; guard-letting outputted the WMPError, giving much more useful info about what caused the error.

Reply:
Only the Swift code should be changed to:

```
store(call.arguments as Int64)
```